apis/Google.Cloud.Logging.Console/Google.Cloud.Logging.Console.Tests/GoogleCloudConsoleFormatterTest.cs

```diff
@@ -31,13 +35,108 @@ public void SimplestLog()
+            var expectedJson = "{\"message\":\"test\",\"severity\":\"INFO\",\"scopes\":[]}\n";
+            var actualJson = writer.ToString();
+
+            Assert.Equal(expectedJson, actualJson);
+        }
+
+
+        [Fact]
+        public void Log_KeyValuePair()
+        {
+            var formatter = CreateFormatter();
+
+            var _state = new ReadOnlyCollection<KeyValuePair<string, object>>(new List<KeyValuePair<string, object>>()
+            {
+                new KeyValuePair<string, object>("A","Sample value 1"),
+                new KeyValuePair<string, object>("B","Sample value 2"),
+            });
+
+            var logEntry = new LogEntry<IReadOnlyCollection<KeyValuePair<string, object>>>(LogLevel.Information, "LogCategory", new EventId(1), _state, exception: null, (state, exception) => state.ToString());
+            var writer = new StringWriter { NewLine = "\n" };
+
+            formatter.Write(logEntry, scopeProvider: null, writer);
+            
+            var actualJson = writer.ToString();
+
+            Assert.Contains("Sample value 1", actualJson);
+            Assert.Contains("Sample value 2", actualJson);
+        }
+
+        [Fact]
+        public void ConsoleLoggerOptions_OptionChange_IsReloaded()
+        {
+            var monitor = new TestFormatterOptionsMonitor<GoogleCloudConsoleFormatterOptions>(new GoogleCloudConsoleFormatterOptions());
+            var formatter = CreateFormatterWithOptionMonitor(monitor);
+            var scopeProvider = new LoggerExternalScopeProvider();
+
+            var logEntry = new LogEntry<string>(LogLevel.Information, "LogCategory", new EventId(1), "test", exception: null, (state, exception) => state);
+            var writer = new StringWriter { NewLine = "\n" };
+            formatter.Write(logEntry, scopeProvider: scopeProvider, writer);
+            var expectedJson = "{\"message\":\"test\",\"severity\":\"INFO\"}\n";
+            var actualJson = writer.ToString();
+
+            Assert.Equal(expectedJson, actualJson);
+
+            monitor.Set(new GoogleCloudConsoleFormatterOptions() { IncludeScopes = true });
+
+            var writerScope = new StringWriter { NewLine = "\n" };
+            formatter.Write(logEntry, scopeProvider: scopeProvider, writerScope);
+            var actualJsonWithScope = writerScope.ToString();
+            var expectedJsonWithScope = "{\"message\":\"test\",\"severity\":\"INFO\",\"scopes\":[]}\n";
+
+            Assert.Equal(expectedJsonWithScope, actualJsonWithScope);
         }
 
         private static GoogleCloudConsoleFormatter CreateFormatter(GoogleCloudConsoleFormatterOptions options = null)
         {
             options ??= new GoogleCloudConsoleFormatterOptions();
             return new GoogleCloudConsoleFormatter(options);
         }
+
+        private static GoogleCloudConsoleFormatter CreateFormatterWithOptionMonitor(IOptionsMonitor<GoogleCloudConsoleFormatterOptions> optionsMonitor = null)
```

Comment:
I'm not sure this is worth breaking out for just one test.

Reply:
We need more tests or call directly constructor on test?

Reply:
Just inline the code into the one place it's used.